jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecs/exec/RunCommandDialog.kt

```diff
@@ -0,0 +1,101 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.ecs.exec
+
+import com.intellij.execution.executors.DefaultRunExecutor
+import com.intellij.execution.runners.ExecutionEnvironmentBuilder
+import com.intellij.openapi.actionSystem.impl.SimpleDataContext
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.tools.Tool
+import com.intellij.tools.ToolRunProfile
+import com.intellij.ui.components.JBTextField
+import com.intellij.ui.layout.panel
+import software.aws.toolkits.jetbrains.core.AwsResourceCache
+import software.aws.toolkits.jetbrains.core.credentials.activeCredentialProvider
+import software.aws.toolkits.jetbrains.core.credentials.activeRegion
+import software.aws.toolkits.jetbrains.services.ecs.ContainerDetails
+import software.aws.toolkits.jetbrains.services.ecs.resources.EcsResources
+import software.aws.toolkits.jetbrains.utils.ui.selected
+import software.aws.toolkits.resources.message
+import javax.swing.DefaultComboBoxModel
+import javax.swing.JComponent
+
+class RunCommandDialog(private val project: Project, private val container: ContainerDetails) : DialogWrapper(project) {
+    private val resourceCache = AwsResourceCache.getInstance()
+    private val tasks = resourceCache.getResourceNow(
+        EcsResources.listTasks(
+            container.service.clusterArn(),
+            container.service.serviceArn()
+        ),
+        project.activeRegion(),
+        project.activeCredentialProvider(),
+        useStale = false,
+        forceFetch = true
+    )
+
+    private val taskList = DefaultComboBoxModel(tasks.toTypedArray())
+    var task = if (tasks.isNotEmpty()) tasks.first() else null
+    private val command = JBTextField()
```

Comment:
We should store the command as a String var and use the UI DSL using `textField(::command)`

Reply:
`command` is now associated with a combobox